File: CS/CustomDocument/Custom DocumentManager/CustomWidgetViewRegistrator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking2010;
using DevExpress.XtraBars.Docking2010.Views;
using DevExpress.XtraBars.Docking2010.Views.Widget;

namespace CustomDocumentApplication {
    class CustomWidgetViewRegistrator : BaseRegistrator {
        public CustomWidgetViewRegistrator()
            : base(null) {
        }
        public CustomWidgetViewRegistrator(WidgetView view)
            : base(view) {
        }
        public override BaseViewPainter CreatePainter() {
            if (View.IsSkinPaintStyle)
                return new WidgetViewSkinPainter(View as WidgetView);
            return new WidgetViewPainter(View as WidgetView);
        }
        public override BaseViewInfo CreateViewInfo() {
            return new WidgetViewInfo(View as WidgetView);
        }
        public override BaseViewHitInfo CreateHitInfo() {
            return new WidgetViewHitInfo(View as WidgetView);
        }
        public override BaseDocument CreateDocument(Control control) {
            return new CustomDocument(View.DocumentProperties as IDocumentProperties);
        }
        protected override Create Register() {
            return delegate(BaseView view) {
                return new CustomWidgetViewRegistrator(view as WidgetView);
            };
        }
        public override DocumentContainer CreateDocumentContainer(BaseDocument document) {
            return new WidgetContainer(document);
        }
        public override Control CreateDocumentsHost(IDocumentsHostOwner owner) {
            return new WidgetsHost(owner);
        }

        new CustomWidgetView View {
            get { return base.View as CustomWidgetView; }
        }
    }
}

File: VB/CustomDocument/Custom DocumentManager/CustomWidgetViewRegistrator.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Docking2010
Imports DevExpress.XtraBars.Docking2010.Views
Imports DevExpress.XtraBars.Docking2010.Views.Widget

Namespace CustomDocumentApplication
	Friend Class CustomWidgetViewRegistrator
		Inherits BaseRegistrator
		Public Sub New()
			MyBase.New(Nothing)
		End Sub
		Public Sub New(ByVal view As WidgetView)
			MyBase.New(view)
		End Sub
		Public Overrides Function CreatePainter() As BaseViewPainter
			If View.IsSkinPaintStyle Then
				Return New WidgetViewSkinPainter(TryCast(View, WidgetView))
			End If
			Return New WidgetViewPainter(TryCast(View, WidgetView))
		End Function
		Public Overrides Function CreateViewInfo() As BaseViewInfo
			Return New WidgetViewInfo(TryCast(View, WidgetView))
		End Function
		Public Overrides Function CreateHitInfo() As BaseViewHitInfo
			Return New WidgetViewHitInfo(TryCast(View, WidgetView))
		End Function
		Public Overrides Function CreateDocument(ByVal control As Control) As BaseDocument
			Return New CustomDocument(TryCast(View.DocumentProperties, IDocumentProperties))
		End Function
		Protected Overrides Function Register() As Create
			Return Function(view As BaseView) New CustomWidgetViewRegistrator(TryCast(view, WidgetView))
		End Function
		Public Overrides Function CreateDocumentContainer(ByVal document As BaseDocument) As DocumentContainer
			Return New WidgetContainer(document)
		End Function
		Public Overrides Function CreateDocumentsHost(ByVal owner As IDocumentsHostOwner) As Control
			Return New WidgetsHost(owner)
		End Function

		Private Shadows ReadOnly Property View() As CustomWidgetView
			Get
				Return TryCast(MyBase.View, CustomWidgetView)
			End Get
		End Property
	End Class
End Namespace
